http/oidc/src/main/java/org/wildfly/security/http/oidc/OidcRequestAuthenticator.java

```diff
@@ -416,4 +484,112 @@ private static boolean hasScope(String scopeParam, String targetScope) {
         }
         return false;
     }
+    private String convertToRequestParameter(URIBuilder redirectUriBuilder, String redirectUri, String state, List<NameValuePair> forwardedQueryParams) throws Exception {
+        redirectUriBuilder.addParameter(SCOPE, OIDC_SCOPE);
+        forwardedQueryParams.add(new BasicNameValuePair(STATE, state));
+        forwardedQueryParams.add(new BasicNameValuePair(REDIRECT_URI, redirectUri));
+        forwardedQueryParams.add(new BasicNameValuePair(RESPONSE_TYPE, CODE));
+        forwardedQueryParams.add(new BasicNameValuePair(CLIENT_ID, deployment.getResourceName()));
+
+        JwtClaims jwtClaims = new JwtClaims();
+        jwtClaims.setIssuer(deployment.getIssuerUrl());
+        jwtClaims.setAudience(deployment.getProviderUrl());
+        for ( NameValuePair parameter: forwardedQueryParams) {
+            jwtClaims.setClaim(parameter.getName(), parameter.getValue());
+        }
+
+        // sign JWT first before encrypting
+        JsonWebSignature signedRequest = signRequest(jwtClaims);
+
+        // Encrypting optional
+        if (deployment.getRequestEncryptAlgorithm() != null && !deployment.getRequestEncryptAlgorithm().isEmpty() &&
+            deployment.getRequestEncryptEncValue() != null && !deployment.getRequestEncryptEncValue().isEmpty()) {
+            return encryptRequest(signedRequest).getCompactSerialization();
+        } else {
+            return signedRequest.getCompactSerialization();
+        }
+    }
+
+    public KeyPair getkeyPair() throws IOException {
+        if (deployment.getClientKeyStoreFile().contains(PROTOCOL_CLASSPATH)) {
```

Comment:
What is the PROTOCOL_CLASSPATH?

Reply:
its a constant with the value Classpath: which can be appended at the front of the path for a keystore file so that if the keystore file is in the target folder, then instead of adding the whole file path, you can just add the path relative to the classpath inside the target folder.

Reply:
Is this used elsewhere?

Reply:
The original loadKeyPairFromKeyStore function can do this too.
On the elytron-oidc-subsystem truststores can be specified using classpath: https://github.com/wildfly/wildfly/blob/main/elytron-oidc-client/src/main/resources/schema/wildfly-elytron-oidc-client_2_0.xsd#L423-L432
So, I thought it could be useful to have. But we can remove it too. 
I haven't used it in examples or for integration tests.